server/src/main/java/org/opensearch/search/query/TopDocsCollectorContext.java

```diff
@@ -254,14 +256,24 @@ private CollapsingTopDocsCollectorContext(
             this.collapseContext = collapseContext;
             this.topDocsCollector = collapseContext.createTopDocs(sort, numHits);
             this.trackMaxScore = trackMaxScore;
+            this.sortByScore = sortAndFormats == null || SortField.FIELD_SCORE.equals(sortAndFormats.sort.getSort()[0]);
 
-            MaxScoreCollector maxScoreCollector = null;
-            if (trackMaxScore) {
+            final MaxScoreCollector maxScoreCollector;
+            if (sortByScore) {
```

Comment:
I’m curious why we don’t retrieve it from the `maxScoreCollector` (as shown in line 271), and instead need to get the maximum score from topDocs.scoreDocs[0]?